### ISSUE TITLE
Add Player Equipment to EntityUpdates

### ISSUE THREAD

**Issue by LarsKoelpin** [markus-wa/cs-demo-minifier#38]
Hey,
I feel like the current Equipment of the Players is missing in the snapshots.

It would be nice, if the EntityUpdates contain the equipment the player currently has in each slot.
Do you think it is much of an effort to implement them / Is there any reason to not implement them?

I gues something like this woudl be nice
```json

{
"tick": 102,
"entityUpdates": [
{
  // ..
  "entityId":1,
  "positions": ["..."],
  "equipment": [
     {"type": 1, "originalString": "", 
     {"type": 101, "originalString": "", },
     {"type": 401, "originalString": "", },
     {"type": 504, "originalString": "", },
     {"type": 504, "originalString": "", },
   ]
  }]
}

```

For example for a player who has
* 2 flashes
* MP7
* P2000

Would it work like this?

Greetings

**Comment by markus-wa** [comment]
sounds reasonable. though I'd like to leave out as much info as possible, ("originalString" etc.).
maybe we should also only set this block if it has changed since the last snapshot to save some bytes.

**Comment by LarsKoelpin** [comment]
In my opinion the byte saving part should be done using gzip etc and is not the scope of this library? (Or maybe add some flags to decide what to serialize)
I think Ease of use beats optimization there.

Would an API APi work like this tho? I’d try implementing an PR if so

**Comment by markus-wa** [comment]
It's always difficult to weigh up the impact on gzip when adding data. Gzip isn't necessarily a magic tool and bigger input -> bigger output. Sometimes it's really good at what it does and there could be little difference between the two or sometimes it's really quite bad.

Either way, we might have some misunderstanding about what `originalString` means: https://pkg.go.dev/github.com/markus-wa/demoinfocs-golang/v2@v2.2.0/pkg/demoinfocs/common?tab=doc#Equipment
If anythin we'd probably want `name` as the attribute, which is `Equipment.Type.String()` (maybe `Equipment.String()` works as well actually)

Could you try comparing the solution with the weapon name in the snapshots to another one where the id to name mappings are in the header instead? Then let me know the byte sizes with and without compression? I'd absolutebly be open if the difference isn't too crazy 😄

**Comment by LarsKoelpin** [comment]
Just to be clear: I don't exactly care about the "orginal String" name. Type  only is fine for me :)

I'm just more advocating a more complete API and let the optimization up to the user and let him decide what he needs and therefore strive for completeness.


The GZIP was aiming at `only set this block if it has changed since the last snapshot to save some bytes.` which implies much more logic on minifying and usage side.
Also I'm talking about the JSON case here ofc, where GZIP defininately makes a difference

**Comment by markus-wa** [comment]
> The GZIP was aiming at only set this block if it has changed since the last snapshot to save some bytes. which implies much more logic on minifying and usage side of the API.

right yeah. that suggestion was probably a bit over the top :sweat_smile:

**Comment by LarsKoelpin** [comment]
Closed with #39
